File: src/main/java/com/gestion_actas/gestion/de/actas/controller/ActasController.java
package com.gestion_actas.gestion.de.actas.controller;

import com.gestion_actas.gestion.de.actas.model.ActasPersonal;
import com.gestion_actas.gestion.de.actas.model.DTO.*;
import com.gestion_actas.gestion.de.actas.services.ActasService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
public class ActasController {
        //implementacion del servicio
        @Autowired
        private ActasService actasService;

        //Obtener detalles
        @GetMapping("/usuarios/{id}")
        public ActasPersonal obtenerUsuario(@PathVariable Long id) {
        return actasService.obtenerPorId(id);
    }

    @GetMapping("/usuarios/detalles")
    public List<DetalleActasDTO> obtenerDetallesSimples(@RequestParam String nroDocumento) {
        return actasService.getActasPorNroDocumento(nroDocumento);
    }


    @GetMapping("usuarios/detalle/{dni}")
    public ResponseEntity<UsuarioDetalleDTO> obtenerPorDni(@PathVariable String dni) {
        UsuarioDetalleDTO dto = actasService.obtenerFormularioPorDni(dni);
        if (dto == null) {
            return ResponseEntity.notFound().build();
        }
        return ResponseEntity.ok(dto);
    }

    //devolver el json de periodos
    @GetMapping("/periodo")
    public List<InicioPeriodosDTO> listarPeriodos() {
        return actasService.listarPeriodos();
    }

    //obtener los usuarios por cada periodo
    @GetMapping("/periodo/{periodoPago}")
    public List<UsuarioResumenDTO> listarUsuariosPorPeriodo(@PathVariable String periodoPago) {
        return actasService.obtenerUsuariosPorPeriodo(periodoPago);
    }
    @GetMapping("/usuarios/lista-unicos")
    public List<ListaUsuariosDTO> listarUsuariosUnicosPorDni() {
        return actasService.listarUsuariosUnicosPorDni();
    }

    /**
     * Endpoint para obtener todos los periodos de pago y id por un DNI.
     * GET /usuarios/periodos/{dni}
     * Devuelve los periodos de pago y el id de cada registro donde aparece el DNI.
     */
    @GetMapping("/usuarios/periodos/{dni}")
    public List<PeriodosDTO> listarPeriodosPorDni(@PathVariable String dni) {
        return actasService.listarPeriodosPorDni(dni);
    }

}

File: Gestion_Actas_Laravel/app/Http/Controllers/ActasController.php
<?php

namespace App\Http\Controllers;

use App\Services\ActasService;
use Illuminate\Http\Request;
use Illuminate\Http\JsonResponse;
use App\Models\ActasPersonal;
use App\Http\Resources\ActaPersonalResource;

class ActasController extends Controller
{
    public function __construct(private ActasService $actasService) {}

    // GET /usuarios/{id} -> ahora con claves camelCase como en Java
    public function obtenerUsuario(int $id): JsonResponse
    {
        $model = ActasPersonal::find($id);
        if (!$model) return response()->json(['error' => 'No encontrado'], 404);
        return response()->json((new ActaPersonalResource($model))->toArray(request()));
    }

    // GET /usuarios/detalles?nroDocumento=...
    public function obtenerDetallesSimples(Request $request): JsonResponse
    {
        $request->validate(['nroDocumento' => 'required|string']);
        $list = $this->actasService->getActasPorNroDocumento($request->get('nroDocumento'));
        return response()->json($list);
    }

    // GET /usuarios/detalle/{dni}
    public function obtenerPorDni(string $dni): JsonResponse
    {
        $dto = $this->actasService->obtenerFormularioPorDni($dni);
        if (!$dto) return response()->json(['error' => 'No encontrado'], 404);
        return response()->json($dto);
    }

    // GET /periodo
    public function listarPeriodos(): JsonResponse
    {
        return response()->json($this->actasService->listarPeriodos());
    }

    // GET /periodo/{periodoPago}
    public function listarUsuariosPorPeriodo(string $periodoPago): JsonResponse
    {
        return response()->json($this->actasService->obtenerUsuariosPorPeriodo($periodoPago));
    }

    // GET /usuarios/lista-unicos
    public function listarUsuariosUnicosPorDni(): JsonResponse
    {
        return response()->json($this->actasService->listarUsuariosUnicosPorDni());
    }

    // GET /usuarios/periodos/{dni}
    public function listarPeriodosPorDni(string $dni): JsonResponse
    {
        return response()->json($this->actasService->listarPeriodosPorDni($dni));
    }

    // GET /descuentos?nro_documento=...
    public function getPeriodosEgresos(Request $request): JsonResponse
    {
        $request->validate(['nro_documento' => 'required|string']);
        $list = $this->actasService->getPeriodosEgresosByNroDocumento($request->get('nro_documento'));
        return response()->json($list);
    }
}